pkg/jx/cmd/create_client_openapi.go

```diff
@@ -0,0 +1,180 @@
+package cmd
+
+import (
+	"go/build"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/jenkins-x/jx/pkg/kube"
+	"github.com/pkg/errors"
+
+	"github.com/jenkins-x/jx/pkg/jx/cmd/templates"
+
+	"github.com/jenkins-x/jx/pkg/log"
+
+	"github.com/jenkins-x/jx/pkg/util"
+
+	"github.com/spf13/cobra"
+	"gopkg.in/AlecAivazis/survey.v1/terminal"
+)
+
+// CreateClientOpenAPIOptions the options for the create client openapi command
+type CreateClientOpenAPIOptions struct {
+	CreateClientOptions
+	Title                string
+	Version              string
+	ReferenceDocsVersion string
+	OpenAPIDependencies  []string
+	OpenAPIGenVersion    string
+	OpenAPIOutputDir     string
+}
+
+var (
+	createClientOpenAPILong = templates.LongDesc(`This command code generates OpenAPI specs for
+	the specified custom resources.
+ 
+`)
+
+	createClientOpenAPIExample = templates.Examples(`
+		# lets generate client docs
+		jx create client openapi
+			--output-package=github.com/jenkins-x/jx/pkg/client \
+			--input-package=github.com/jenkins-x/pkg-apis \
+			--group-with-version=jenkins.io:v1
+			--version=1.2.3
+			--title=Jenkins X
+		
+		# You will normally want to add a target to your Makefile that looks like:
+
+		generate-openapi:
+			jx create client openapi
+				--output-package=github.com/jenkins-x/jx/pkg/client \
+				--input-package=github.com/jenkins-x/jx/pkg/apis \
+				--group-with-version=jenkins.io:v1
+				--version=${VERSION}
+				--title=${TITLE}
+		
+		# and then call:
+
+		make generate-openapi
+`)
+)
+
+// NewCmdCreateClientOpenAPI creates the command
+func NewCmdCreateClientOpenAPI(f Factory, in terminal.FileReader, out terminal.FileWriter,
+	errOut io.Writer) *cobra.Command {
+	o := &CreateClientOpenAPIOptions{
+		CreateClientOptions: CreateClientOptions{
+			CreateOptions: CreateOptions{
+				CommonOptions: CommonOptions{
+					Factory: f,
+					In:      in,
+					Out:     out,
+					Err:     errOut,
+				},
+			},
+		},
+	}
+
+	cmd := &cobra.Command{
+		Use:     "openapi",
+		Short:   "Creates OpenAPI specs for Custom Resources",
+		Long:    createClientOpenAPILong,
+		Example: createClientOpenAPIExample,
+
+		Run: func(cmd *cobra.Command, args []string) {
+			o.Cmd = cmd
+			o.Args = args
+			err := o.Run()
+			CheckErr(err)
+		},
+	}
+
+	wd, err := os.Getwd()
+	if err != nil {
+		log.Warnf("Error getting working directory for %v\n", err)
+	}
+
+	openAPIDependencies := []string{
```

Comment:
Do we need to keep this in sync with the dependencies from `go.mod`?

Reply:
Yes, but I couldn't figure out how to do this. I've added a TODO for now.

Reply:
In particular I could not for the life of me work out how to get any `go` command to output the resolved version of a requirement.

Reply:
`go mod graph` was the answer :-) Added a parser for it and loading version from there now.